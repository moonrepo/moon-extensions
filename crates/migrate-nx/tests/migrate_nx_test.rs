use moon_pdk_test_utils::{create_extension, ExecuteExtensionInput};
use starbase_sandbox::{assert_snapshot, create_empty_sandbox, create_sandbox};
use std::fs;

mod migrate_nx {
    use super::*;

    #[test]
    fn converts_root_files() {
        let sandbox = create_sandbox("root");
        let plugin = create_extension("test", sandbox.path());

        plugin.execute_extension(ExecuteExtensionInput {
            args: vec![],
            context: plugin.create_context(sandbox.path()),
        });

        assert!(!sandbox.path().join("nx.json").exists());
        assert!(!sandbox.path().join("workspace.json").exists());
        assert!(sandbox.path().join(".moon/tasks/node.yml").exists());
        assert!(sandbox.path().join(".moon/workspace.yml").exists());

        assert_snapshot!(fs::read_to_string(sandbox.path().join(".moon/tasks/node.yml")).unwrap());
        assert_snapshot!(fs::read_to_string(sandbox.path().join(".moon/workspace.yml")).unwrap());
    }

    mod projects {
        use super::*;

        #[test]
        fn uses_defaults() {
            let sandbox = create_empty_sandbox();
            let plugin = create_extension("test", sandbox.path());

            plugin.execute_extension(ExecuteExtensionInput {
                args: vec![],
                context: plugin.create_context(sandbox.path()),
            });

            assert_snapshot!(
                fs::read_to_string(sandbox.path().join(".moon/workspace.yml")).unwrap()
            );
        }

        #[test]
        fn inherits_layout() {
            let sandbox = create_empty_sandbox();
            sandbox.create_file(
                "nx.json",
                r#"
{
  "workspaceLayout": {
    "appsDir": "applications",
    "libsDir": "libraries"
  }
}"#,
            );

            let plugin = create_extension("test", sandbox.path());

            plugin.execute_extension(ExecuteExtensionInput {
                args: vec![],
                context: plugin.create_context(sandbox.path()),
            });

            assert_snapshot!(
                fs::read_to_string(sandbox.path().join(".moon/workspace.yml")).unwrap()
            );
        }
    }
}
