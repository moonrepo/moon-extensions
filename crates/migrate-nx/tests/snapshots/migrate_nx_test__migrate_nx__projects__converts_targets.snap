---
source: crates/migrate-nx/tests/migrate_nx_test.rs
expression: "fs::read_to_string(sandbox.path().join(\"moon.yml\")).unwrap()"
---
language: javascript
platform: node
tasks:
  build:
    command: esbuild
    args:
    - --main
    - app
    - --tsConfig
    - $workspaceRoot/app/tsconfig.app.json
    - --outputPath
    - $workspaceRoot/dist/app
  create-script:
    command: mkdir -p apps/frontend/scripts && touch apps/frontend/scripts/my-script.sh && chmod +x apps/frontend/scripts/my-script.sh
    platform: system
  dev:
    command: next server
    args:
    - --buildTarget
    - acme:build
    - --no-turbo
    - --dev
  e2e:
    command: cypress
    args:
    - --cypressConfig
    - $workspaceRoot/apps/app-e2e/cypres.config.ts
    - --devServerTarget
    - my-react-app:serve
    - --testingType
    - e2e
  ls-project-root:
    command: ls apps/frontend/src
    platform: system
  noop:
    command: noop
  run-script:
    command: npm run build-my-project
  serve:
    command: angular dev-server
  serve.development:
    extends: serve
    args:
    - --buildTarget
    - my-app:build:development
  serve.production:
    extends: serve
    args:
    - --buildTarget
    - my-app:build:production
  test:
    command: jest
    args:
    - --passWithNoTests
    - --jestConfig
    - $workspaceRoot/libs/my-lib/jest.config.ts
